chain-impl-mockchain/src/leadership/genesis/mod.rs

```diff
@@ -100,13 +107,15 @@ impl GenesisLeaderSelection {
                             total: total_stake,
                         };
 
-                        let _ = vrfeval::verify(
-                            percent_stake,
-                            &pool_info.initial_key.vrf_public_key,
-                            &self.epoch_nonce,
-                            block_header.block_date().slot_id,
-                            &genesis_praos_proof.vrf_proof,
-                        );
+                        let _ = vrfeval::Verifier {
+                            key_stake: percent_stake,
```

Comment:
Good. A builder-style API might be even better, unless all of these parameters are mandatory.

Reply:
They are mandatory and not defaultable, at least in current usage context